draft-ietf-taps-arch.md

```diff
@@ -182,12 +182,14 @@ There are key differences between the architecture of the Transport Services sys
 
 Originally, sockets presented a blocking interface for establishing connections and transferring data. However, most modern applications interact with the network asynchronously. When sockets are presented as an asynchronous interface, they generally use a try-and-fail model. If the application wants to read, but data has not yet been received from the peer, the call to read will fail. The application then waits and can try again later.
 
-All interaction with a Transport Services system is expected to be asynchronous, and use an event-driven model unlike sockets {{events}}. For example, if the application wants to read, its call to read will not fail, but will deliver an event containing the received data once it is available.
+All interaction with a Transport Services system is expected to be asynchronous, and use an event-driven model unlike sockets (see {{events}}). For example, if the application wants to read, its call to read will not return immediately, but will deliver an event containing the received data once it is available. Error handling is also asynchronous; a failure to send results in an asynchronous send error as an event. 
```

Comment:
Not sure if we want to say it won't "return immediately"... that's not quite true for async callbacks depending on the language. The function will return immediately, but have a delayed callback.

How about "its request/call to read will not complete immediately"?